### PR TITLE
feat: whatsapp desktop app

### PR DESCRIPTION
**Pull request by harshv5094**
## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Documentation update
- [ ] Refactoring
- [ ] Hotfix
- [ ] Security patch
- [ ] UI/UX improvement

## Description
[ZapZap](https://github.com/rafatosta/zapzap) is an open source `WhatsApp` desktop client. This script allows to install ZapZap in your system

## Testing
For arch, I've tested on my own system. In case of other distro's, I've set Flatpak as default option.

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

**Comment by lj3954** [comment]
Perhaps it would be better to call the entry 'ZapZap', with the description specifying that it's a client for WhatsApp.

File: core/tabs/applications-setup/communication-apps/whatsapp-setup.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    ;;
+  *)
+    . ../setup-flatpak.sh
+    printf "%b\n" "${YELLOW} Installing Whatsapp(Zap-Zap by rtosta) Flatpak ${RC}"
+    flatpak install flathub com.rtosta.zapzap
+    printf "%b\n" "${GREEN}WhatsApp(Zap-Zap by rtosta) is installed. Search ZapZap to start WhatsApp${RC}"
+    return
+    ;;
+  esac
+}
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/communication-apps/whatsapp-setup.sh:27]:
```suggestion
  if ! command_exists com.rtosta.zapzap && ! command_exists zapzap; then
    printf "%b\n" "${YELLOW}Installing Whatsapp(Zap-Zap by rtosta)${RC}"
    case "$PACKAGER" in
      pacman)
        "$AUR_HELPER" -S --needed --noconfirm zapzap
        ;;
      *)
        . ../setup-flatpak.sh
        flatpak install flathub com.rtosta.zapzap
        ;;
    esac
  else
    printf "%b\n" "${GREEN}WhatsApp(Zap-Zap by rtosta) is already installed.${RC}"
  fi
}
```

**Review by jeevithakannan2 — CHANGES_REQUESTED** [review]

**Review by jeevithakannan2 — APPROVED** [review]

**Comment by jeevithakannan2** [comment]
Try to keep consistency with other scripts. This is not a big deal but linutil uses 4 space or tabs instead of 2 spaces.

**Review by jeevithakannan2 — APPROVED** [review]

**Comment by harshv5094** [comment]
> Try to keep consistency with other scripts. This is not a big deal but linutil uses 4 space or tab instead of 2 spaces.
Okay, I think it has to do with my lazyvim. I'll make sure this won't happen again

**Comment by adamperkowski** [comment]
> Perhaps it would be better to call the entry 'ZapZap', with the description specifying that it's a client for WhatsApp.

I agree.

**Comment by harshv5094** [comment]
okay I'll change it

> > Perhaps it would be better to call the entry 'ZapZap', with the description specifying that it's a client for WhatsApp.
> 
> I agree.

File: core/tabs/applications-setup/tab_data.toml [hunk 2]
```diff
@@ -78,6 +78,12 @@ description = "Telegram is a cloud-based messaging app known for its speed and s
 script = "communication-apps/telegram-setup.sh"
 task_list = "I"
 
+[[data.entries]]
+name = "ZapZap (Whatsapp Client)"
+description = "ZapZap is a whatsapp desktop client for linux users, It is an open source project developed by rafatosta."
```
Comment (adamperkowski) [inline core/tabs/applications-setup/tab_data.toml:83]:
```suggestion
description = "ZapZap is an open source whatsapp desktop client for Linux users developed by rafatosta."
```

File: core/tabs/applications-setup/communication-apps/zapzap-setup.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh -e
+
+. ../../common-script.sh
+
+installWhatsapp() {
+  printf "%b\n" "${YELLOW}The app we are installing is Zap-Zap, an Open Source Desktop Client for Linux${RC}"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/communication-apps/zapzap-setup.sh:6]:
```suggestion
installZapZap() {
```

File: core/tabs/applications-setup/communication-apps/zapzap-setup.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  if ! command_exists com.rtosta.zapzap && ! command_exists zapzap; then
+    printf "%b\n" "${YELLOW}Installing Whatsapp(Zap-Zap by rtosta)${RC}"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/communication-apps/zapzap-setup.sh:8]:
```suggestion
    printf "%b\n" "${YELLOW}Installing Zap-Zap...${RC}"
```

File: core/tabs/applications-setup/communication-apps/zapzap-setup.sh [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      pacman)
+        "$AUR_HELPER" -S --needed --noconfirm zapzap
+        ;;
+      *)
+        . ../setup-flatpak.sh
+        flatpak install flathub com.rtosta.zapzap
+        ;;
+    esac
+  else
+    printf "%b\n" "${GREEN}WhatsApp(Zap-Zap by rtosta) is already installed.${RC}"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/communication-apps/zapzap-setup.sh:19]:
```suggestion
    printf "%b\n" "${GREEN}Zap-Zap is already installed.${RC}"
```

File: core/tabs/applications-setup/communication-apps/zapzap-setup.sh [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  fi
+}
+
+checkEnv
+installWhatsapp
```
Comment (adamperkowski) [inline core/tabs/applications-setup/communication-apps/zapzap-setup.sh:24]:
```suggestion
installZapZap
```

**Review by adamperkowski — CHANGES_REQUESTED** [review]

**Comment by harshv5094** [comment]
Hey @adamperkowski, can you help me clearing this shell formatting check? It's returning error code 1

**Comment by jeevithakannan2** [comment]
Run `shfmt -i 4 -ci path/to/file.sh`

File: core/tabs/applications-setup/communication-apps/zapzap-setup.sh [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+installZapZap() {
+  if ! command_exists com.rtosta.zapzap && ! command_exists zapzap; then
+  printf "%b\n" "${YELLOW}Installing Zap-Zap...${RC}"
+    case "$PACKAGER" in
+      pacman)
+        "$AUR_HELPER" -S --needed --noconfirm zapzap
+        ;;
+      *)
+        . ../setup-flatpak.sh
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/communication-apps/zapzap-setup.sh:13]:
```suggestion
        checkFlatpak
```

**Review by jeevithakannan2 — CHANGES_REQUESTED** [review]

File: core/tabs/applications-setup/tab_data.toml [hunk 8]
```diff
@@ -84,6 +84,18 @@ description = "Sublime Text is a fast, lightweight, and customizable text editor
 script = "Developer-tools/sublime-setup.sh"
 task_list = "I"
 
+[[data.entries]]
+name = "
+(Whatsapp Client)"
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/tab_data.toml:89]:
```suggestion
name = "ZapZap"
```

File: core/tabs/applications-setup/tab_data.toml [hunk 9]
```diff
@@ -84,6 +84,18 @@ description = "Sublime Text is a fast, lightweight, and customizable text editor
...
+description = "ZapZap is an open source whatsapp desktop client for Linux users developed by rafatosta."
+script = "communication-apps/zapzap-setup.sh"
+task_list = "I"
+
+[[data.entries]]
+name = "Zoom"
+description = "Zoom is a widely-used video conferencing platform that allows users to host virtual meetings, webinars, and online collaboration with features like screen sharing and recording."
+script = "communication-apps/zoom-setup.sh"
+
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/tab_data.toml:98]:
```suggestion
```

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**